### ISSUE TITLE
Error on database migration on 1.6.2 upgrade

### ISSUE THREAD

**Issue by xam-ps** [InvoicePlane/InvoicePlane#1147]
When Invoiceplane is upgraded from 1.6.1 to 1.6.2 (beta 2) the database migration is causing an error and migration can't be continued.

## Expected Behavior
Database migration should be successful

## Current Behavior
Database migration is causing an error and ip is broken

## Screenshots
![grafik](https://github.com/user-attachments/assets/93677ce2-0566-4a55-986a-49bad3d30b62)

## Possible Solution
When table is deleted manually, the migration can be continued, but this should not be the recommended solution. This should be fixed in the migration script somehow.

## Steps to Reproduce
1. Replace old (1.6.1) files with new (1.6.2) files
2. Go to ` /index.php/setup/upgrade_tables`
3. Click through the migration wizard

**Comment by sudwebdesign** [comment]
Hi @xam-ps this error is really curious, normally the `ip_login_log` table is [created at ip v1.5.12](https://github.com/InvoicePlane/InvoicePlane/blob/development/application/modules/setup/sql/035_1.5.12.sql).

I check, Fresh db install of ip v1.6.1 and upgrade to [dev (1.6.2)](https://github.com/InvoicePlane/InvoicePlane/commit/d61968c87990574f0e29830c5b076a020486817d) and all it's would be fine.

**Comment by nielsdrost7** [comment]
I still want to make a doublecheck

**Comment by sudwebdesign** [comment]
@nielsdrost7 you're absolutely right :+1:

**Comment by xam-ps** [comment]
@sudwebdesign Got this instance of ip already running for a longer time. Could be indeed a problem that was already caused in an earlier upgrade.

**Comment by xam-ps** [comment]
@sudwebdesign 
Just double checked: When I'm running `/index.php/setup/upgrade_tables` on my 1.6.1 instance, I get the same error as well. So I guess this was not introduced with 1.6.2

Thank you for investigating. Closing the issue.

**Comment by nielsdrost7** [comment]
hey @xam-ps I'll keep it open for now.
I want to be 100% sure that some of the other users aren't getting this error as well.

Check Discord, I've sent you a message

**Comment by AeroBytesNL** [comment]
@xam-ps 

I'm working on this bug. 
Could you check for me what the value is off `ENVIRONMENT` in `index.php`?

**Comment by AeroBytesNL** [comment]
My bad for the commit/PR mess!

**Comment by xam-ps** [comment]
@AeroBytesNL 
where is it/should it be stored? ipconfig.php? In the index.php there should not be any ENV-Variables or am I mistaken?

**Comment by sudwebdesign** [comment]
Hi, @AeroBytesNL this code never return errors. 

It preferable to use negative comparisons 
and replace `continue` by  `$this->errors[] = $this->db->_error_message();`

```php
            if (($this->db->error()['code']) !== 1050 || ($this->db->error()['code'] !== 1060)) {
                $this->errors[] = $this->db->_error_message();
            }
```
But never return this 2 error codes.

Maybe it's a bug of `IPv1.6.1` see @xam-ps preview comment https://github.com/InvoicePlane/InvoicePlane/issues/1147#issuecomment-2558375310

Note: If IP support php 5x, maybe get a parse error, modify like this:
```php
            $error = $this->db->error();
            if (($error['code']) !== 1050 || ($error['code'] !== 1060)) {
                $this->errors[] = $this->db->_error_message();
            }
```
See [Mdl_setup line 57](https://github.com/InvoicePlane/InvoicePlane/blob/70211b1647f1cf374969104ec9077bd69aa2f605/application/modules/setup/models/Mdl_setup.php#L57)

**Comment by AeroBytesNL** [comment]
@xam-ps 
There is an `define('ENVIRONMENT, 'some value')' in 'index.php'. At least on my installation

**Comment by sudwebdesign** [comment]
Other Idea:
In `application/modules/setup/sql/035_1.5.12.sql`
add ```DROP TABLE IF EXISTS `ip_login_log`;```

```sql
DROP TABLE IF EXISTS `ip_login_log`;
CREATE TABLE `ip_login_log` (
    `login_name` varchar(100),
    `log_count` int DEFAULT 0,
    `log_create_timestamp` datetime,
    PRIMARY KEY (login_name)
)
```
But loose old login logs.

**Comment by xam-ps** [comment]
@AeroBytesNL 
`define('ENVIRONMENT', isset($_SERVER['CI_ENV']) ? $_SERVER['CI_ENV'] : 'development');`
This is the line in my index.php

**Comment by nielsdrost7** [comment]
> In application/modules/setup/sql/035_1.5.12.sql
add DROP TABLE IF EXISTS `ip_login_log`;

Let's not mess with old migrations

> IMO this should be defined in ipconfig.php and not somewhere in the middle of index.php 🤔

> This is also not mentioned in the [documentation](https://wiki.invoiceplane.com/en/1.6/getting-started/installation).

You are correct, but let's not deal with that 1 day before the release

**Comment by AeroBytesNL** [comment]
> @AeroBytesNL `define('ENVIRONMENT', isset($_SERVER['CI_ENV']) ? $_SERVER['CI_ENV'] : 'development');` This is the line in my index.php
> 
> IMO this should be defined in ipconfig.php and not somewhere in the middle of index.php 🤔
> 
> This is also not mentioned in the [documentation](https://wiki.invoiceplane.com/en/1.6/getting-started/installation).

Yeah fair haha.

**Comment by AeroBytesNL** [comment]
> Hi, @AeroBytesNL this code never return errors.
> 
> It preferable to use negative comparisons and replace `continue` by `$this->errors[] = $this->db->_error_message();`
> 
> ```
>             if (($this->db->error()['code']) !== 1050 || ($this->db->error()['code'] !== 1060)) {
>                 $this->errors[] = $this->db->_error_message();
>             }
> ```
> 
> But never return this 2 error codes.
> 
> Maybe it's a bug of `IPv1.6.1` see @xam-ps preview comment [#1147 (comment)](https://github.com/InvoicePlane/InvoicePlane/issues/1147#issuecomment-2558375310)
> 
> Note: If IP support php 5x, maybe get a parse error, modify like this:
> 
> ```
>             $error = $this->db->error();
>             if (($error['code']) !== 1050 || ($error['code'] !== 1060)) {
>                 $this->errors[] = $this->db->_error_message();
>             }
> ```
> 
> See [Mdl_setup line 57](https://github.com/InvoicePlane/InvoicePlane/blob/70211b1647f1cf374969104ec9077bd69aa2f605/application/modules/setup/models/Mdl_setup.php#L57)

That it not returns those two errors was by design.
So it ignored those two errors. 
Tho i should as @nielsdrost7 said on review

> and replace `continue` by `$this->errors[] = $this->db->_error_message();`

I like you're solution, i shall try to remember it! 
I will implement it in a bit! Thanks for the feedback.

**Comment by nielsdrost7** [comment]
I'll keep it open until after the release

**Comment by nielsdrost7** [comment]
Solved by [releasing version 1.6.2](https://github.com/InvoicePlane/InvoicePlane/releases/tag/v1.6.2)
